### ISSUE TITLE
minTickSize fails when set to 'year' and not enough data

### ISSUE THREAD

**Issue by GoogleCodeExporter** [josejaner/flot#195]
```
I've been using minTickSize to set proper values for ticks depending on if 
data needs to be grouped by date, month, year, etc. When set to 
[1, 'month'] and there are only dates of 3 months in the plot, it shows 
ticks for all three months. Years however requires 5 years of data to be 
present; otherwise it doesn't show any ticks or labels for them.

See example at http://evgenevich.com/random/time.html demonstrating 
difference between 3 and 5 years of data with both [1, 'year'] and 
[1, 'month'].

As I mentioned before, you can have data for only 3 months and using 
[1, 'month'] will show ticks for 3 months, but with years this fails 
unless you have 5 years of data. Thus I'm wondering if this is a bug in 
how the year setting is processed/rendered?
```

Original issue reported on code.google.com by `NEvgenev...@gmail.com` on 23 Jul 2009 at 3:39

**Comment by GoogleCodeExporter** [comment]
```
It certainly might be. However, I can't access your example. Sorry for not 
getting
back to you earlier. If you still have the problem, feel free to look into the 
code
in Flot that does the calculation and see if you can find a problem.
```

Original comment by `olau%iol...@gtempaccount.com` on 7 Oct 2009 at 4:24

**Comment by GoogleCodeExporter** [comment]
```
Confirmed by removing some data from the Time example; with minTickSize of one 
year, ticks don't show unless there are at least five years of data.
```

Original comment by `dnsch...@gmail.com` on 4 Jun 2012 at 8:41
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Issue 574 has been merged into this issue.
```

Original comment by `dnsch...@gmail.com` on 4 Jun 2012 at 9:18

**Comment by GoogleCodeExporter** [comment]
```
Mark: here's another time-related issue.  I've confirmed the problem; it's easy 
to reproduce exactly as described.
```

Original comment by `dnsch...@gmail.com` on 6 Jun 2012 at 3:25

**Comment by GoogleCodeExporter** [comment]
```
I can reproduce this, although I determined the cut-off to be about 5.5 years, 
not 5 years. When I changed the xaxis min the "nineties" example in time.html 
from 1990-01-01 to 1995-01-01 I got no labels; the earliest min date for which 
labels appeared was 1994-06-29.

I've identified the source of the problem. In tick generation, it appears there 
is a special case for years. There is some calculation, which I don't 
understand at the moment, that yields a tick size of less than 1 when there is 
less than about 5.5 years of data. There is special handling for tickSize < 1 
when the unit is months, but none for years, and tick generation stalls because 
incrementing the tick (d.setFullYear(d.getFullYear() + tickSize)) results in 
the same date when tickSize < 1.

This is actually sometimes a problem even when minTickSize *isn't* 
specified--it defaults to years when the date range is above 5 years, so from 5 
to 5.5 years there are no labels.

The main fix is to allow tick sizes of less than 1 when the unit is years. This 
is already supported when the unit is months, and I think it makes sense for 
years.

In addition, the special tick-size adjustment for unit of years should be 
ignored if minTickSize is greater than the calculated size.

Neither of these are too hard; should be done soon, and I'll add the 5-year 
scenario to the time.html example, since I guess it's as close to a test suite 
as we have. :)
```

Original comment by `mrcote` on 8 Jun 2012 at 4:50

**Comment by GoogleCodeExporter** [comment]
```
Actually forget support for tick sizes of less than one year--we already 
support 1, 2, 3, and 6 months, and we never supported arbitrary tick sizes. 
Sole fix is to respect minTickSize when unit is years.
```

Original comment by `mrcote` on 8 Jun 2012 at 5:18

**Comment by GoogleCodeExporter** [comment]
```
Fixed in commit 1f484ed2931f598d89d47a06e38af32bc46b8e5d. flot was actually 
always ignoring minTickSize when it was given in years, and the automatic tick 
calculation for years was also broken when the time range was less than about 
5.5 years. Both are fixed, and I added an example to verify.
```

Original comment by `mrcote` on 14 Jun 2012 at 9:02
- Changed state: **Fixed**
